src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,82 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final double RESIZE_FACTOR = 0.75;
+    private Node<K, V>[] storage;
+    private int size;
+
+    public MyHashMap() {
+        storage = (Node<K, V>[]) new Node[DEFAULT_SIZE];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> existingNode = getNodeByKey(key);
+        if (existingNode != null) {
+            existingNode.value = value;
+            return;
+        }
+        ensureCapacity();
+        insertNode(new Node<>(key, value));
+        size++;
     }
```

Comment:
I don't like the organization of that method because you have split your logic in harm of the performance by duplicating some parts like getting bucket index both in `insertNode()` and `getNodeByKey()`.
So some part of that should be inside here
\+ `insertNode()` was not really necessary to have it separate since you are not reusing it anywhere